src/renderer/src/hooks/useBeforeLogin.tsx

```diff
@@ -176,6 +178,12 @@ const useRemoveTemporaryApplicationData = () => {
 }
 
 export const useBeforeLogin = () => {
+  const { language } = useLanguageSelector()
+
+  useEffect(() => {
+    i18next.changeLanguage(language.value)
+  }, [language])
```

Comment:
Follow the same pattern by creating a custom hook called `useLanguageChange` inside this file